### ISSUE TITLE
Dependent dimensions

### ISSUE THREAD

**Issue by rpep** [ioam/holoviews#839]
Currently, as far as I can find, there's no way to make Dimension values vary given another parameter. This would be a useful option in this situation:

Looking at slices through 3-D simulation data on an L x L x 2L block. Have two HoloViews dimensions:

• hv.Dimension('Dimension', values=['x', 'y', 'z'])

Ideally, therefore, you'd like the values for a coordinate dimension to vary between [0, L] if the x or y axis is selected, and [0, 2L] if z is selected. Currently, you just have to set:

• hv.Dimension('Coordinate, values=np.linspace(0, 2*L, 10))

and make sure the function which generates the image from the data can gracefully handle being out of range.

**Comment by jlstevens** [comment]
It sounds like you are looking for dependent sliders for ``DynamicMap`` which I agree would be very nice to have!

There is a new 'streams' system which was are currently in the process of introducing that I believe will go some way to tackling this problem. Hopefully we'll be able to come back to you with an example once more of this system is in place...

**Comment by ryanpepper** [comment]
@jlstevens That's exactly it :) 

Great, look forward to it :)

**Comment by philippjfr** [comment]
So this is really a duplicate of https://github.com/ioam/holoviews/issues/301 and my opinion is that this should be expressed through param and its widget extensions (i.e. parambokeh and paramnb). This will ensure that complex dependencies between parameters can be expressed without adding to the complexity of the default HoloViews widgets.
